File: app/src/main/java/com/example/myapplication/networks/ApiEndPointService.java
package com.example.myapplication.networks;


import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Url;

public interface ApiEndPointService {
    @GET
    Observable<String> getSearchAnimeData(
            @Url String getAllAnimeDataUrl1
    );

    @GET
    Observable<String> getGenreAnimeData(
            @Url String getAllAnimeDataUrl2
    );

    @GET
    Observable<String> getNewReleaseAnimeData(
            @Url String getAllAnimeDataUrl3
    );

    @GET
    Observable<String> getWatchAnimeData(
            @Url String getAllAnimeDataUrl4
    );

    @GET
    Observable<String> getNewReleaseMangaData(
            @Url String getAllMangaDataUrl
    );

    @GET
    Observable<String> getDiscoverMangaData(
            @Url String getAllMangaDataUrl
    );

    @GET
    Observable<String> getReadMangaData(
            @Url String getAllMangaDataUrl
    );

}

File: app/src/main/java/com/example/myapplication/networks/ApiEndPointService.kt
package com.example.myapplication.networks


import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Url

interface ApiEndPointService {
    @GET
    fun getSearchAnimeData(
            @Url getAllAnimeDataUrl1: String
    ): Observable<String>

    @GET
    fun getGenreAnimeData(
            @Url getAllAnimeDataUrl2: String
    ): Observable<String>

    @GET
    fun getNewReleaseAnimeData(
            @Url getAllAnimeDataUrl3: String
    ): Observable<String>

    @GET
    fun getWatchAnimeData(
            @Url getAllAnimeDataUrl4: String
    ): Observable<String>

    @GET
    fun getNewReleaseMangaData(
            @Url getAllMangaDataUrl: String
    ): Observable<String>

    @GET
    fun getDiscoverMangaData(
            @Url getAllMangaDataUrl: String
    ): Observable<String>

    @GET
    fun getReadMangaData(
            @Url getAllMangaDataUrl: String
    ): Observable<String>

}
